db/db_impl/db_impl_files.cc

```diff
@@ -193,11 +193,14 @@ void DBImpl::FindObsoleteFiles(JobContext* job_context, bool force,
       }
     }
 
+    IOOptions io_opts;
+    io_opts.property_bag["skip_directories"] = "true";
```

Comment:
It might also be nice to create a IOOptions & IOOptions::EnableProperty(name) method that sets the named property.  You might want to consider having a default value of "true" to allow the property to be set.  Comparably, a std::string IOOptions::GetProperty(name, default_value) might be useful as well.